labgraph/lsl_node/tests/test_lsl_node.py

```diff
@@ -0,0 +1,253 @@
+        data = f.read()
+    recieved_data = set(data.strip(DATA_DELIMITER).split(DATA_DELIMITER))
+    assert len(recieved_data) > 0
+    assert len(samples) == len(recieved_data)
+
+
+@local_test
+def test_lsl_sender_node() -> None:
+    class LSLSenderGraph(Graph):
+        MY_SOURCE: MySource
+        MY_SINK: LSLSenderNode
+
+        config: LSLSenderConfig
+
+        def setup(self) -> None:
+            self.MY_SOURCE.configure(MySourceConfig())
+            self.MY_SINK.configure(self.config)
+
+        def connections(self) -> Connections:
+            return ((self.MY_SOURCE.TOPIC, self.MY_SINK.topic),)
+
+    output_filename = get_test_filename()
+    graph = LSLSenderGraph()
+    graph.configure(LSLSenderConfig(
+        name='Test',
+        type='TEST',
+        n_channels=NUM_MESSAGES,
+        unique_identifier='12345QE'
+    ))
+    runner = LocalRunner(module=graph)
+    p = Process(
+        target=recv_samples_from_lsl, args=(output_filename,)
+    )
+    p.start()
+    runner.run()
+    p.join()
+
+    with open(output_filename, "r") as f:
+        data = f.read()
+
+    recieved_data = set(data.strip(DATA_DELIMITER).split(DATA_DELIMITER))
+    assert len(recieved_data) > 0
+    assert len(samples) == len(recieved_data)
+
+
+@local_test
+def test_lsl_send_and_poll() -> None:
+    class LSLGraphConfig(Config):
+        output_filename: str
+
+    class LSLGraph(Graph):
+        DF_SOURCE: MySource
+        LSL_SENDER: LSLSenderNode
+        LSL_POLLER: LSLPollerNode
+        DF_SINK: MySink
+
+        def setup(self) -> None:
+            self.DF_SOURCE.configure(MySourceConfig(should_terminate=True))
+            self.LSL_SENDER.configure(
+                LSLSenderConfig(
+                    name='Test',
+                    type='EEG',
+                    n_channels=NUM_MESSAGES,
+                    unique_identifier='12345QE'
+                )
+            )
+            self.LSL_POLLER.configure(
+                LSLPollerConfig(
+                    type='EEG'
+                )
+            )
+            self.DF_SINK.configure(
+                MySinkConfig(output_filename=self.config.output_filename)
+            )
+
+        def connections(self) -> Connections:
+            return(
+                (self.DF_SOURCE.TOPIC, self.LSL_SENDER.topic),
+                (self.LSL_POLLER.topic, self.DF_SINK.TOPIC)
+            )
+
+        def process_modules(self) -> Sequence[Module]:
+            return (self.DF_SOURCE, self.LSL_SENDER, self.LSL_POLLER,
+                    self.DF_SINK)
+
+    output_filename = get_test_filename()
+    graph = LSLGraph()
+    graph.configure(
+        LSLGraphConfig(
+            output_filename=output_filename
+        )
+    )
+    runner = ParallelRunner(graph=graph)
+    runner.run()
+
+    with open(output_filename, "r") as f:
+        data = f.read()
+    recieved_data = set(data.strip(DATA_DELIMITER).split(DATA_DELIMITER))
+    assert len(recieved_data) > 0
```

Comment:
shall we use len(received_data) == NUM_MESSAGES instead
Please also check whether there are any typos in the PR.

Reply:
Done